File: API/Models/Purchase.cs
using static API.Models.PurchaseItem;

namespace API.Models;

public class Purchase
{
    public long id { get; set; }
    public int userID { get; set; }
    public string userName { get; set; } = "";
    public DateTime orderDate { get; set; }
    public decimal total { get; set; }
    public int productID { get; set; }
    public string productName { get; set; } = "";
    public List<Product>? products { get; set; }

    public class PurchaseInsert
    {
        public int userID { get; set; } = -1;
        public DateTime orderDate { get; set; } = DateTime.UtcNow;
        public List<PurchaseItemInsert> items { get; set; } = [];
    }

    public class PurchaseUpdate
    {
        public int id { get; set; } = -1;
        public int userID { get; set; } = -1;
        public int productID { get; set; } = -1;
        public DateTime orderDate { get; set; } = DateTime.UtcNow;
        public decimal total { get; set; } = 0;
    }
}

File: Frontend/src/app/core/models/Purchase.ts
export class Purchase {
    id: number = -1;
    userID: number = -1;
    productID: number = -1;
    productName?: string;
    userName?: string;
    orderDate: Date = new Date();
    total: number = 0;
    products: { name: string; quantity: number }[] = [];
}

export class PurchaseInsert {
    userID: number = -1;
    items: { productID: number; quantity: number }[] = [];
}


export class PurchaseUpdate {
    id: number = -1;
    userID: number = -1;
    productID: number = -1;
    orderDate: Date = new Date();
    total: number = 0;
}